data-serving/data-service/src/controllers/case.ts

```diff
@@ -75,13 +76,37 @@ export const list = async (req: Request, res: Response): Promise<void> => {
     }
 };
 
+// Finds sources with an existing URL in the request body and replace them
+// for the ones that already exist.
+// TODO: Consider calling find() with all the IDs at once if multiple sources
+// per case become the norm to avoid roundtrips to mongo.
+// TODO: Support more than just URL to fetch existing sources.
+async function fillInExistingSources(req: Request): Promise<void> {
+    const sources = req.body.sources;
+    // No sources will automatically fail at validation time.
+    if (!sources) {
+        return;
+    }
+    for (let i = 0; i < sources.length; i++) {
```

Comment:
nit: `for (let requestSource of sources)`?

Reply:
well I need to update the req.body.sources[i], if I don't get an index I can't update it right?
unless there's something like *requestSource = *source like in Go that does a deep copy of the pointed variable.